pkgs/unified_analytics/lib/src/asserts.dart

```diff
@@ -122,21 +145,38 @@ void checkBody(Map<String, Object?> body) {
     // GA4 Limitation:
     // User property names must be 24 characters or fewer
     if (key.length > 24) {
-      throw AnalyticsException('Limit user property names to 24 chars or less');
+      throw AnalyticsException('Limit user property names to 24 chars or less',
+          moreInfo: 'The user property: $key has a value that is too long');
     }
 
     // GA4 Limitation:
     // User property values must be 36 characters or fewer
     final userPropValue = value['value'];
     if (userPropValue is String && userPropValue.length > 36) {
       throw AnalyticsException(
-          'Limit user property values to 36 chars or less');
+        'Limit user property values to 36 chars or less',
+        moreInfo: 'The value for the user property: $key which has '
+            'a value: $value is too long',
+      );
     }
   }
 }
 
 class AnalyticsException implements Exception {
   final String message;
 
-  AnalyticsException(this.message);
+  /// Optional field to pass if there is specific details
+  /// that can be included with the error message.
+  final String? moreInfo;
```

Comment:
do we need another field for this? The caller could just concatenate the moreInfo value to their `message` field, right?

Reply:
it feels like adding this optional field is adding complexity here when the implementation is just concatenating the second string, if present, and prefacing it with "Details:\n", when I feel like the code would be more obvious if this formatting is done at the point the exception is thrown.

Reply:
I kept them different because in `asserts_test.dart` I am looking for a specific message, such as `Event param name first char must be alphabetic char`.

If I concatenated them, the test cases become harder to maintain if I change the test data

Reply:
But I suppose I can use `startsWith` but I felt an exact match on the string was more robust

Reply:
if your'e using `package:test` you can use the `contains()` matcher

Reply:
> I kept them different because in asserts_test.dart I am looking for a specific message, such as Event param name first char must be alphabetic char.

> If I concatenated them, the test cases become harder to maintain if I change the test data

Can you interpolate the test data value into the expectation string?

Reply:
I agree robust tests is a worthy goal, but preferably not at the cost of public API complexity

Reply:
> Can you interpolate the test data value into the expectation string?

I was actually testing that out right now, and agreed on the reducing complexity and keeping the tests robust

Reply:
So this an example of where I am throwing in `asserts.dart`

```dart
      throw AnalyticsException(
        'Event name first char must be alphabetic char',
        moreInfo: 'Event name: $eventName must begin with a valid character',
      );
```

Do you think this would look better instead for the API?

```dart
      throw AnalyticsException(
        'Event name first char must be alphabetic char\n'
        'Details\nEvent name: $eventName must begin with a valid character',
      );
```

Reply:
Yeah, I think formatting the string from the caller would be preferable.

Reply:
Gotcha, just refactored everything